### ISSUE TITLE
1.8.2: shell freezes, requiring Ctrl-C to continue

### ISSUE THREAD

**Issue by hexsprite** [meteor/meteor#10803]
Problem observed using MacOS running in iTerm and Terminal.app

I will often find that seemingly random but with regularity that the shell will just freeze and I will need to hit Ctrl-C to restart the session. Thankfully Meteor keeps the shell contents so I don't lose all my work but it's quite annoying to do this several times per debugging session.

Not sure how to reproduce this at this point. Hoping to get some pointers on how to debug something like this...

**Comment by hexsprite** [comment]
I was able to reproduce this in a bare project.

I did:

`fs = require('fs')`

Then typed `fs.` and hit TAB to autocomplete.

I hit ESC and repeated it several times until it finally hung.

**Comment by fred8617** [comment]
![image](https://user-images.githubusercontent.com/35049517/69837246-bcc37600-1288-11ea-873d-92b13a87c9a6.png)
![image](https://user-images.githubusercontent.com/35049517/69837260-ca78fb80-1288-11ea-82cf-4a9ee8adeba9.png)
maybe cancel this checkbox

**Comment by hexsprite** [comment]
thanks @fred8617 I'm on Mac. What does that checkbox say/do?

**Comment by hexsprite** [comment]
Still an issue


Jordan

**Comment by hexsprite** [comment]
still an issue

**Comment by claudiosdc** [comment]
I am also experiencing this issue ever since I have upgraded my development meteor environment from version 1.8.1 to version 1.9. I am currently using Meteor version 1.10.1 on macOS 10.15.4 and the issue persists.

I was doing some research and found out that the issue was introduced when the "shell-client.js" module has been converted to TypeScript generating the new "shell-client.ts" module. Here is what I did:

1. I ran my meteor app (using meteor version 1.10.1) normally;
2. I cloned meteor's git repository, and used it to run meteor's shell to connect to my meteor app;
3. From meteor's shell, I forced the error (by repeatedly listing the properties of one of my global objects);
4. Then I repeated the process, only this time replacing the "shell-client.ts" module on the local meteor's git repository with the old "shell-client.js" module;
5. Now I was not able to force the error anymore.
